### DISCUSSION TITLE
Simple Todo Management System

### DISCUSSION THREAD

**Discussion by AlxGolubev** [outline/outline#6935] [Ideas]
The magic of any knowledge database is truly realized when all information about a topic or project is consolidated in one place. Having documentation, meeting notes, plans, and statuses all in one space enhances team focus and simplifies access management across various tools and services.

While project management might not be Outline's primary focus, meeting notes often end with a list of action items. Since we can already create tasks on a page and see task completion status, why not take it a step further? Imagine being able to assign tasks to team members, set deadlines, and add tags directly within Outline. This would allow us to keep todos like "call John about the next release" in the same place where meeting notes are stored.

I believe you'd agree that having a separate service for simple todos might be overkill. But envision if we could consolidate all our todos across different documents and then summarize them by person or tag in one view/document. This would make it much easier and clearer for users to manage their simple plans, todos, and action items without cluttering a full-fledged project management system.

There is a lot of examples of how it could work.
1. Confluence - you can create action items on page, mention different people and dates and then on the other page you can list all todos by label/person - https://confluence.atlassian.com/doc/add-assign-and-view-tasks-590260030.html
2. The other but even better example is Obsidian Dataview - https://blacksmithgu.github.io/obsidian-dataview/annotation/metadata-tasks/

Here's a simple example of meeting notes
<img width="776" alt="Screenshot 2024-05-23 at 10 03 53" src="https://github.com/outline/outline/assets/2353443/98f0c64c-6b80-45b4-b2d8-5b5f477ab6a2">

And here' a page with all tasks that has ERPSample tag
<img width="822" alt="Screenshot 2024-05-23 at 10 04 37" src="https://github.com/outline/outline/assets/2353443/b752742d-b600-4f79-9094-785bbb342533">

Right now tasks completion are calculated on the fly during page rendering, but I guess we can make a step further here and keep all documentation in one place

**Comment by tommoor** [comment]
A first step would be to write this as a plugin,

- Introduce a `Task` model that is controlled through checklists on a document
- `TaskProcessor` similar to `BacklinksProcessor` would parse documents on edits to ensures tasks are always uptodate with docs
- `tasks` API router with `list` endpoint (create and delete handled by checklists in first instance)
- UI to view tasks, check them off, delete, notifications etc.

There are a lot of edge cases to think of, this is not a small task and may well take a few months to get to prod.

**Reply by AlxGolubev** [reply]
Thank you for your response. I understand the complexity of the task, but I believe implementing it would be highly beneficial.

In the future, when I have available resources, I can allocate them to start working on this initiative. At that point, I will reach out to discuss how I can assist your team in delivering this project, should you find the concept appealing.

Looking forward to our potential collaboration.

**Comment by ucola** [comment]
We use Outline for checklists (similar to to-dos) and assign tasks to users using the @mention method. It would be helpful to have an overview of all documents where I am mentioned.
